### DISCUSSION TITLE
Use .mdx instead of .md

### DISCUSSION THREAD

**Discussion by nik-rev** [TheOdinProject/curriculum#28631] [Ideas]
MDX allows for custom markdown syntax and using react components in markdown

For example, we wouldnt have to write `<div class="lesson-note lesson-note-warning>` but could use custom syntax, e.g. the same one that github uses

```
> [!NOTE] 
> Useful information that users should know, even when skimming content.
```

This would also unlock other potential, we could create custom components when we need to enhance user experience. e.g. there could be a quiz component

If this is accepted the following would have to be done:
1. In the website repo, create a react component that will render MDX
2. `react-rails` can then render the component to display in `.erb` files

**Comment by mao-sz** [comment]
Thanks for the suggestion. Had a chat with some team members and a lot more would need to be done than has been suggested, including but not limited to:

- converting every lesson file to .mdx, *then* converting every note box/assignment panel etc. to the appropriate custom syntax
- overhauling how we lint the curriculum repo
- overhauling how we convert markdown to html since Kramdown doesn't support .mdx either
- reintroducing react-rails after it was removed when the website repo switched from react components to Hotwire for maintainability reasons

There are likely other big considerations that just haven't sprung to mind yet. Overall, there isn't quite enough benefit to warrant that much of a change across the board, so it's not something we'd likely want to explore.
